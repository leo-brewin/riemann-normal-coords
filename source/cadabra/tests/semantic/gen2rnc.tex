\def\Date{19 Jan 2024}
% \def\FileID{file:}

\documentclass[12pt]{cdblatex}

\begin{document}

\section*{\jobname}

\CdbSetup{action=hide}

\begin{cadabra}
   import shared

   import cdblib

   term00A = cdblib.get ('check000','expected/gen2rnc.json')
   term01A = cdblib.get ('check001','expected/gen2rnc.json')
   term02A = cdblib.get ('check002','expected/gen2rnc.json')
   term03A = cdblib.get ('check003','expected/gen2rnc.json')
   term04A = cdblib.get ('check004','expected/gen2rnc.json')

   term00B = cdblib.get ('check000','output/gen2rnc.json')
   term01B = cdblib.get ('check001','output/gen2rnc.json')
   term02B = cdblib.get ('check002','output/gen2rnc.json')
   term03B = cdblib.get ('check003','output/gen2rnc.json')
   term04B = cdblib.get ('check004','output/gen2rnc.json')

   diff000 = shared.check (term00A,term00B)   # cdb (diff000,diff000)
   diff001 = shared.check (term01A,term01B)   # cdb (diff001,diff001)
   diff002 = shared.check (term02A,term02B)   # cdb (diff002,diff002)
   diff003 = shared.check (term03A,term03B)   # cdb (diff003,diff003)
   diff004 = shared.check (term04A,term04B)   # cdb (diff004,diff004)

\end{cadabra}

\begin{dgroup*}
   \Dmath*{ \cdb*{diff000} }
   \Dmath*{ \cdb*{diff001} }
   \Dmath*{ \cdb*{diff002} }
   \Dmath*{ \cdb*{diff003} }
   \Dmath*{ \cdb*{diff004} }
\end{dgroup*}

\end{document}
