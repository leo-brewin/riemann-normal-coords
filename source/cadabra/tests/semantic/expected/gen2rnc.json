{
  "check000": "x^{a}",
  "check001": "x^{b} x^{c} \\Gamma^{a}_{b c}",
  "check002": "x^{b} x^{c} x^{d} (\\Gamma^{a}_{b e} \\Gamma^{e}_{c d} + \\partial_{b}(\\Gamma^{a}_{c d}))",
  "check003": "x^{b} x^{c} x^{d} x^{e} (2\\Gamma^{a}_{b f} \\partial_{c}(\\Gamma^{f}_{d e}) + \\Gamma^{a}_{f g} \\Gamma^{f}_{b c} \\Gamma^{g}_{d e} + \\Gamma^{f}_{b c} \\partial_{f}(\\Gamma^{a}_{d e}) + \\partial_{b c}(\\Gamma^{a}_{d e}))",
  "check004": "x^{b} x^{c} x^{d} x^{e} x^{f} (-4\\Gamma^{a}_{b g} \\Gamma^{g}_{c h} \\Gamma^{h}_{d i} \\Gamma^{i}_{e f} + 2\\Gamma^{a}_{b g} \\Gamma^{g}_{c h} \\partial_{d}(\\Gamma^{h}_{e f}) + 3\\Gamma^{a}_{b g} \\Gamma^{g}_{h i} \\Gamma^{h}_{c d} \\Gamma^{i}_{e f}-6\\Gamma^{a}_{b g} \\Gamma^{h}_{c d} \\partial_{e}(\\Gamma^{g}_{f h}) + 6\\Gamma^{a}_{b g} \\Gamma^{h}_{c d} \\partial_{h}(\\Gamma^{g}_{e f}) + 9\\Gamma^{a}_{b g} \\partial_{c d}(\\Gamma^{g}_{e f}) + 4\\Gamma^{a}_{g h} \\Gamma^{g}_{b c} \\Gamma^{h}_{d i} \\Gamma^{i}_{e f} + 13\\Gamma^{a}_{g h} \\Gamma^{g}_{b c} \\partial_{d}(\\Gamma^{h}_{e f})-4\\Gamma^{g}_{b c} \\Gamma^{h}_{d g} \\partial_{e}(\\Gamma^{a}_{f h}) + \\Gamma^{g}_{b c} \\Gamma^{h}_{d g} \\partial_{h}(\\Gamma^{a}_{e f}) + 2\\partial_{b}(\\Gamma^{a}_{c g}) \\partial_{d}(\\Gamma^{g}_{e f}) + 7\\partial_{g}(\\Gamma^{a}_{b c}) \\partial_{d}(\\Gamma^{g}_{e f}) + 3\\Gamma^{g}_{b c} \\Gamma^{h}_{d e} \\partial_{f}(\\Gamma^{a}_{g h}) + 3\\Gamma^{g}_{b c} \\Gamma^{h}_{d e} \\partial_{g}(\\Gamma^{a}_{f h})-3\\Gamma^{g}_{b c} \\partial_{d e}(\\Gamma^{a}_{f g}) + 6\\Gamma^{g}_{b c} \\partial_{d g}(\\Gamma^{a}_{e f}) + 3\\partial_{b c d}(\\Gamma^{a}_{e f}))"
}
